openlibrary/core/fulltext.py

```diff
@@ -17,16 +16,17 @@ def fulltext_search_api(params):
     search_select = search_endpoint + '?' + urllib.parse.urlencode(params, 'utf-8')
 
     try:
-        json_data = urllib.request.urlopen(search_select, timeout=30).read()
+        response = requests.get(search_select, timeout=30)
         logger = logging.getLogger("openlibrary.inside")
         logger.debug('URL: ' + search_select)
-    except:
+        response.raise_for_status()
+    except Exception:
```

Comment:
Let's tighten the error to `requests.HTTPError` since this is what `response.raise_for_status()` will raise.

https://realpython.com/the-most-diabolical-python-antipattern/